packages/yswaps/contracts/TradeFactory/TradeFactoryExecutor.sol

```diff
@@ -16,44 +16,49 @@ import './TradeFactoryPositionsHandler.sol';
 
 interface ITradeFactoryExecutor {
   event SyncTradeExecuted(
+    SyncTradeExecutionDetails _tradeExecutionDetails,
     address indexed _strategy,
-    address indexed _swapper,
-    address _tokenIn,
-    address _tokenOut,
-    uint256 _amountIn,
-    uint256 _maxSlippage,
-    uint256 _receivedAmount
+    uint256 _receivedAmount,
+    address indexed _swapper
   );
 
-  event AsyncTradeExecuted(
-    address indexed _strategy,
-    address indexed _tokenIn,
-    address indexed _tokenOut,
-    address _swapper,
-    uint256 _amountIn,
-    uint256 _minAmountOut,
-    uint256 _receivedAmount
-  );
+  event AsyncTradeExecuted(AsyncTradeExecutionDetails _tradeExecutionDetails, uint256 _receivedAmount, address _swapper);
+
+  event MultipleAsyncTradeExecuted(AsyncTradeExecutionDetails[] _tradesExecutionDetails, uint256[] _receivedAmount, address _swapper);
 
   error InvalidAmountOut();
 
+  struct SyncTradeExecutionDetails {
+    address _tokenIn;
+    address _tokenOut;
+    uint256 _amountIn;
+    uint256 _maxSlippage;
+  }
+
+  struct AsyncTradeExecutionDetails {
+    address _strategy;
+    address _tokenIn;
+    address _tokenOut;
+    uint256 _amount;
+    uint256 _minAmountOut;
+  }
```

Comment:
This was the best way I found to have a struct shared through the `execute` and `execute multi`